test/e2e/scheduling/nvidia-gpus.go

```diff
@@ -179,3 +182,98 @@ var _ = SIGDescribe("[Feature:GPUDevicePlugin]", func() {
 		testNvidiaGPUs(f)
 	})
 })
+
+func testNvidiaGPUsJob(f *framework.Framework) {
+	_ = SetupNVIDIAGPUNode(f, false)
+	completions := int32(5)
+	By("Starting GPU job")
+	makeCudaAdditionDevicePluginTestJob(f, completions)
+
+	job, err := jobutil.GetJob(f.ClientSet, f.Namespace.Name, "cuda-add")
+	Expect(err).NotTo(HaveOccurred())
+
+	// make sure job is running by waiting for its first pod to start running
+	err = jobutil.WaitForAllJobPodsRunning(f.ClientSet, f.Namespace.Name, job.Name, 1)
+	Expect(err).NotTo(HaveOccurred())
+
+	numNodes, err := framework.NumberOfRegisteredNodes(f.ClientSet)
+	Expect(err).NotTo(HaveOccurred())
+	nodes, err := framework.CheckNodesReady(f.ClientSet, numNodes, framework.NodeReadyInitialTimeout)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Recreating nodes")
+	gce.RecreateNodes(f.ClientSet, nodes)
+	By("Done recreating nodes")
+
+	By("Waiting for gpu job to finish")
+	err = jobutil.WaitForJobFinish(f.ClientSet, f.Namespace.Name, job.Name)
+	Expect(err).NotTo(HaveOccurred())
+	By("Done with gpu job")
+
+	Expect(job.Status.Failed).To(BeZero(), "Job pods failed during node recreation: %v", job.Status.Failed)
+
+	verifyJobNCompletions(f, completions)
+}
+
+func makeCudaAdditionDevicePluginTestJob(f *framework.Framework, completions int32) {
```

Comment:
I moved the two functions from `test/e2e/upgrades/nvidia-gpu.go` to `test/e2e/scheduling/nvidia-gpus.go` since importing `upgrades` within `scheduling` creates an import cycle.